smart_open/http.py

```diff
@@ -114,3 +133,118 @@ def readinto(self, b):
             return 0
         b[:len(data)] = data
         return len(data)
+
+
+class SeekableBufferedInputBase(BufferedInputBase):
```

Comment:
Maybe better use a more specific name (in case if this applied only for HTTP)?

Reply:
IMHO, this is already part of the http module, so everything in there is specific to http a priori, so the name is fine. We use a similar naming scheme in the S3 module as well.